lib/system.nim

```diff
@@ -2498,7 +2498,7 @@ template `^^`(s, i: untyped): untyped =
 template `[]`*(s: string; i: int): char = arrGet(s, i)
 template `[]=`*(s: string; i: int; val: char) = arrPut(s, i, val)
 
-proc `[]`*[T, U](s: string, x: HSlice[T, U]): string {.inline.} =
+proc `[]`*[T: Ordinal, U: Ordinal](s: string, x: HSlice[T, U]): string {.inline.} =
```

Comment:
```suggestion
proc `[]`*[T, U: Ordinal](s: string, x: HSlice[T, U]): string {.inline.} =
```

ditto elsewhere
it should work, given that this works:
```nim
when true:
  proc fn[T, U: Ordinal](a: T, b: U) = discard
  fn(1u8, 2i16)
```

Reply:
Yeah, I just want it more explicit.
But it's ok.